src/file/file.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { fileModel } from './file.model';
+
+// const readdir = util.promisify(fs.readdir);
+// const unlink = util.promisify(fs.unlink);
+const expect: Chai.ExpectStatic = chai.expect;
+chai.use(chaiAsPromised);
+
+const TOTAL_FILES: number = 3;
+const newName: string = 'changeName';
+let testFiles: IFile[];
+
+describe(`File Logic`, () => {
+
+  beforeEach(async () => {
+    // Remove uploadsTEST folder
+    await fs.remove(`${config.storage}`);
+
+    // Remove files from DB
+    const removeCollectionPromises = [];
+    for (const i in mongoose.connection.collections) {
+      removeCollectionPromises.push(mongoose.connection.collections[i].remove({}));
+    }
+    await Promise.all(removeCollectionPromises);
+
+    // Create files in Folder and DB
+    testFiles = createFiles(TOTAL_FILES);
+    await fileController.create(testFiles);
+  });
+
+  describe('#getById', () => {
+    it('Should return a file by its id', async () => {
+      const file: IFile = await fileController.findById(testFiles[0]._id);
+      expect(testFiles[0].equals(file)).to.be.true;
+    });
+  });
+
+  describe('#getByDate', () => {
+    it('Should get all files that were created before now', async () => {
+      const toDate = new Date(Date.now());
+      const ret = await fileController.findByDate(null, toDate.toISOString());
+      expect(ret.length).to.be.equal(TOTAL_FILES);
+    });
+    it('Should return that were created from now', async () => {
+      const fromDate = new Date(Date.now());
+      const ret = await fileController.findByDate(fromDate.toISOString());
+      expect(ret.length).to.be.equal(0);
+    });
+    it('Should get all files created between yesterday and today', async () => {
+      const fromDate = new Date();
+      fromDate.setDate(fromDate.getDate() - 1);
+      const toDate = new Date(Date.now());
+      const ret = await fileController.findByDate(fromDate.toISOString(), toDate.toISOString());
+      expect(ret.length).to.be.equal(TOTAL_FILES);
+    });
+  });
+
+  describe('#getAll', () => {
+    it(`Should return a collection with ${TOTAL_FILES} files`, async () => {
+      const filesReturned: IFile[] = await fileController.getFiles();
+      expect(filesReturned).to.not.be.empty;
+      expect(filesReturned).to.have.lengthOf(testFiles.length);
+    });
+  });
+
+  describe('#update', () => {
+    it('Should update all of the names', async () => {
+      for (let i: number = 0; i < testFiles.length; i++) {
+        await fileController.update(testFiles[i]._id, { _id: testFiles[i]._id, fileName: newName });
+      }
+      const updatedFile: IFile = await fileController.findById(testFiles[0]._id);
+      expect(updatedFile.fileName).to.be.equal(newName);
+    });
+    // it('Should throw exception when trying to update a non-existent file', async () => {
+    //   await expect(fileController.update('non_existent_id', { fileName: 'ErrorName' }))
+    //     .to.eventually.be.rejectedWith('File doesnt exist');
+    // });
```

Comment:
Unnecessary comments.
If you want to do the test later, make it a pending test. Otherwise, delete this.